File: dsa/python3/bubble_sort.py
# /usr/bin/env python3


def bubble_sort(arr):
    for _ in range(0, len(arr)):
        for j in range(0, len(arr) - 1):
            if arr[j] > arr[j + 1]:
                tmp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = tmp
    return arr


if __name__ == "__main__":
    print(bubble_sort([4, 2, 1, 4, 10, 6, 1, 8, 9]))
    print(bubble_sort([-4, 2, 1, 4, 10, -6, 1, 8, 9]))
    print(bubble_sort([-4]))
    print(bubble_sort([]))

File: dsa/cpp/bubble_sort.cpp

#include <cmath>
#include <iostream>
#include <vector>

void bubble_sort(std::vector<int>& array)
{
  for (int i = 0; i < array.size(); ++i) {
    for (int j = 0; j < array.size(); ++j) {
      if (array[j] > array[j + 1]) {
        int tmp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = tmp;
      }
    }
  }
}

std::ostream& operator<<(std::ostream& os, const std::vector<int>& arr)
{
  for (const auto& item : arr) {
    std::cout << item << " ";
  }
  std::cout << std::endl;
  return os;
}

int main()
{
  std::vector<int> array = {0, 3, 2, 5, 6, 8, 1, 9, 4, 2, 1, 2, 9, 6, 4, 1, 7, -1, -5, 23, 6, 2, 35, 6, 3, 32};
  std::cout << array << std::endl;
  bubble_sort(array);
  std::cout << array << std::endl;

  return 0;
}
